### PR TITLE
Add js parser for angular

### PR DESCRIPTION
**Pull request by bretfourbe**
Add basic js parser for angular : extract related domains and paths.
Add brotli module in requirements to support br content-encoding, this will help to detect more js files.

### CODE REVIEW

File: wapitiCore/net/jsparser_angular.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        for path in data_found:
+            if path and "http" not in path and path not in path_found:
+                path_found.append(path)
+
+        # re.findall(r"(?:^\w+$|(?:https?:\/)?[\w\.\/%]*(?:\/|%2F|%2f)(?:[\w:?%#~$][\w\-\.@?=%&:/~#]*[\w=/~]))", values)
+
+        # catch full urls
+        # https://www.i2tutorials.com/match-urls-using-regular-expressions-in-python/
+        urls = \
+            re.findall(r"https?:\/\/(?:[\w\-_]+(?:(?:\.[\w\-_]+)+))(?:[\w\-\.,@?^=%&:/~\+#]*[\w\-\@?^=%&/~\+#])?", data)
```
Comment (jinxka) [inline wapitiCore/net/jsparser_angular.py:78]: `r"https?:\/\/[^\"'\\ )]+"`

**Review by jinxka — CHANGES_REQUESTED** [review]

File: Pipfile [hunk 2]
```diff
@@ -18,6 +18,7 @@ yaswfp = ">=0.9.3"
 six = ">=1.15.0"
 browser-cookie3 = "==0.11.4"
 importlib_metadata = "==3.7.2"
+brotli = ">=1.0.9"
```
Comment (devl00p) [inline Pipfile:21]: I don't see where you are using this but if the objective is being able to decode brotli encoded HTTP requests it seems we should use `httpx[brotli]` as dependency

see https://www.python-httpx.org/#installation

File: wapitiCore/net/jsparser_angular.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            url: The absolute URL of the JS file on the server
+            data: The content of the JS file as a string
+        """
+        self.links = []
+
+        parts = urlparse(url)
+        netloc = parts.netloc
+        self.target_domain = netloc
+
+        if netloc and len(netloc.split('.')) > 2:
```
Comment (devl00p) [inline wapitiCore/net/jsparser_angular.py:25]: You should not extract domain by splitting on dots, it won't work for .co.uk suffixes. Instead use `tld.get_fld` which is already used elsewhere in Wapiti.

Also you should check that parts.hostname is not an IP address.

File: wapitiCore/net/jsparser_angular.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            netloc = '.'.join(netloc.split('.')[1:])
+
+        self.domain = netloc
+        self.scheme = parts.scheme
+
+        if self.domain:
```
Comment (devl00p) [inline wapitiCore/net/jsparser_angular.py:31]: Is there any possible case where `self.domain` is not set ?

**Review by devl00p — CHANGES_REQUESTED** [review]

File: Pipfile (hunk 2 above)
Comment (bretfourbe) [reply Pipfile]: `requests` need `brotli` to decode br encoded responses. Many js files are br encoded, that is why i need it. But now we are using `httpx` so we should use `http[brotli]` as you said.

File: wapitiCore/net/jsparser_angular.py (hunk 4 above)
Comment (bretfourbe) [reply wapitiCore/net/jsparser_angular.py:34]: Only if we have an invalid url

File: Pipfile (hunk 2 above)
Comment (bretfourbe) [reply Pipfile]: But it appears that `httpx[brotli]` does not work on debian... and `brotli` works

Comment (devl00p) [reply Pipfile:21]: Indeed I checked with Ubuntu and openSUSE and `pip install brotli` seems to be enough

**Review by devl00p — APPROVED** [review]

**Pull request merged**